### DISCUSSION TITLE
[question] Changing security without node removal (keeping location/name and ID)

### DISCUSSION THREAD

**Discussion by mazmar** [zwave-js/zwave-js-ui#1973] [Q&A]
Is there an easier way of changing security on existing device? I kinda understand that the device has to be excluded and then included in order for S2 to work but this means all node ids will change and that will messup with my hass.io and I will need to organise all switches again?

**Comment by robertsLando** [comment]
Thre is a step by step guide written by @blhoward2 but I cannot find the link to it. Maybe colud be worth adding it to docs? @blhoward2

**Comment by blhoward2** [comment]
https://github.com/zwave-js/node-zwave-js/issues/3227 is I think what you mean but it's a bit different. The steps are: shutdown zwavejs, reset, turn on, check failed node, replace failed node. 

Others have also said that you can use a second stick to exclude it, if you have one, in lieu of these.

**Comment by scyto** [comment]
@blhoward2 i assume that means reset the zwave end point device not zwave js right?

i have a battery aeotec usb stick i use just for exclusions, so sounds like i can shutdown zwavejs, exclude the node, restart zwavejs, then do a failed node replacement?

**Reply by blhoward2** [reply]
Yes, but you must have HA shutdown until after the inclusion if you're using HA. Otherwise it will nuke the naming info when it receives notification of the new node.

**Comment by blhoward2** [comment]
Correct.

**Comment by kpine** [comment]
If you use a separate z-wave controller to exclude a node, you don't need to shutdown zwave-js.

If you are using the HA Z-Wave JS integration, and release 2021.9, you'll want to shutdown HA so it doesn't see the node removal and therefore remove the device and all its entities. 2021.10 is supposed to be able to detect a node replacement, in that case you don't need to shutdown HA (I haven't tried it).

**Reply by kpine** [reply]
> 2021.10 is supposed to be able to detect a node replacement, in that case you don't need to shutdown HA (I haven't tried it).

Follow-up: 2021.10 did not correctly fix the node replacement problem, so at this time it is still recommended to shutdown HA to preserve the devices.
